Fix email loading

samaradel opened the pull request:
### Description

- The email input was disabled whenever `mnemonicInput?.status !== ValidatorStatus.Valid`, which included when there was no mnemonic at all
- The validation rules were always applied, even when the field should be disabled
- When the mnemonic was cleared, the email state wasn't properly reset

### Changes

[Screencast from 03-08-25 14:49:20.webm](https://github.com/user-attachments/assets/3e3742b7-1acd-4e48-b732-00ac091402f4)

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/4109 

### Tested Scenarios

- When a valid mnemonic: email populated, field disabled, validation disabled
- When mnemonic is deleted: Email cleared, field enabled, validation enabled but not triggered
- When an invalid mnemonic is pasted: Email cleared, field enabled, validation enabled but not triggered
- Check if the user interacts with the enabled email field: Normal validation behavior resumes

### Documentation PR

For UI changes, Please provide the Documentation PR on [info_grid](https://github.com/threefoldtech/info_grid)

### To consider

#### Preliminary Checks:

- [x] Preliminary Checks
  - Does it completely address the issue linked?
  - What about edge cases?
  - Does it meet the specified acceptance criteria?
  - Are there any unintended side effects?
  - Does the PR adhere to the team's coding conventions, style guides, and best practices?
  - Does it integrate well with existing features?
  - Does it impact the overall performance of the application?
  - Are there any bottlenecks or slowdowns?
  - Has it been optimized for efficiency?
  - Has it been adequately tested with unit, integration, and end-to-end tests?
  - Are there any known defects or issues?
  - Is the code well-documented?
  - Are changes to documentation reflected in the code?

#### UI Checks:

- [x] UI Checks
  - If a UI design is provided/ does it follow it?
  - Does every button work?
  - Is the data displayed logical? Is it what you expected?
  - Does every validation work?
  - Does this interface feel intuitive?
  - What about slow network? Offline?
  - What if the gridproxy/graphql/chain is failing?
  - What would a first time user have a hard time navigating here?

#### Code Quality Checks:

- [x] Code Quality Checks
  - Code formatted/linted? Are there unused imports? .. etc
  - Is there redundant/repeated code?
  - Are there conditionals that are always true or always false?
  - Can we write this more concisely?
  - Can we reuse this code? If yes, where?
  - Will the changes be easy to maintain and update in the future?
  - Can this code become too complex to understand for other devs?
  - Can this code cause future integration problems?

### Testing Checklist

- [x] Does it Meet the specified acceptance criteria.
- [x] Test if changes can affect any other functionality.
- [ ] Tested with unit, integration, and end-to-end tests.
- [ ] Tested the un-happy path and rollback scenarios.
- [ ] Documentation updated to meet the latest changes.
- [ ] Check automated tests working successfully with the changes.
- [ ] Can be covered by automated tests.

### General Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstring
- [x] Screenshots/Video attached (needed for UI changes)

Commit:
Fix email loading

0oM4R reviewed (commented):
after creating a new account without touching the email account the email validation got triggered, also after adding mnemonic the email will try to load stored email, if none will show the error, 
<img width="1008" height="813" alt="image" src="https://github.com/user-attachments/assets/6e7d18b3-2c11-4535-8994-d1665d373768" />

Commit:
- Fix Email input validation
- Disable all input in there is no mnemonic
- Prevent email input loading if it is a new account

0oM4R commented on packages/playground/src/components/profile_manager/ConnectWallet.vue:
```diff
@@ -318,7 +318,29 @@ const validateMnemonicInput = async (input: string) => {
+    // Check if twin exists first
+    try {
+      await grid!.twins.get_my_twin_id();
+      // Twin exists, so get email with loading state
+      loadEmail.value = true;
+      try {
+        getEmail(grid!);
+      } finally {
+        loadEmail.value = false;
+      }
```
why did you remove the load email handler from getEmail? 
also the old logic was handle getEmail errors, so no need to handle it here in nested try

samaradel replied:
Well, it was fire even though the mnemonic is new (new account), which is useless

0oM4R replied:
<img width="483" height="296" alt="image" src="https://github.com/user-attachments/assets/6bf05678-6a79-475f-86fb-3b7c3c69e558" />

i don't think it was firing any errors. 
also in your new code `await grid!.twins.get_my_twin_id();` if this failed so it will not trigger load email as it will go to the catch block

0oM4R reviewed (changes requested):
After adding a non-active mnemonic, the email field is disabled and can't proceed
<img width="1002" height="800" alt="image" src="https://github.com/user-attachments/assets/075e9f32-04c9-4b51-8490-d70f46f25844" />

0oM4R commented on packages/playground/src/components/profile_manager/ConnectWallet.vue:
```diff
@@ -124,16 +125,18 @@
-          :disabled="creatingAccount || connecting || loadEmail || mnemonicInput?.status !== ValidatorStatus.Valid"
+          :disabled="isEmailDisabled"
           autocomplete="off"
+          @blur="() => emailInput?.validate()"
+          @update:model-value="() => emailInput?.validate()"
```
I think this is the actual behavior of the input validator. Why do we need to specify that?

samaradel replied:
After handling the disabled part, it doesn't work, so I handled it this way

Commit:
Enable email field if the mnemonic is valid

0oM4R commented on packages/playground/src/components/profile_manager/ConnectWallet.vue:
```diff
@@ -124,16 +125,18 @@
         v-model="password"
         mode="Create"
-        :disabled="creatingAccount || connecting"
+        :disabled="creatingAccount || connecting || !isMnemonicValid"
```
Why do we disable those if the mnemonic is empty? 
The idea behind disabling the email field is to avoid overwriting the entered email after getting the stored email. This case is on password fields, so I don't think we should disable them

samaradel replied:
if we don't have a valid mnemonic, why do we need it enabled? It's useless

0oM4R commented on packages/playground/src/components/profile_manager/ConnectWallet.vue:
```diff
@@ -152,7 +155,7 @@
               ...validationProps,
             }"
-            :disabled="creatingAccount || connecting"
+            :disabled="creatingAccount || connecting || !isMnemonicValid"
```
please check https://github.com/threefoldtech/tfgrid-sdk-ts/pull/4327/files#r2265191703

0oM4R commented on packages/playground/src/components/profile_manager/ConnectWallet.vue:
```diff
@@ -263,27 +290,65 @@ function reloadValidation() {
+    return {
+      message: "Mnemonic or Hex Seed is required.",
+    };
+  }
```
Checking if the mnemonic is empty is done by the required validator, I think we will never reach this

0oM4R reviewed (commented):
I think there are many unnecessary changes, we can add to mnemonic field somthing like 
` @update:model-value="() => email ? (email = '') : null"` 
and will reset the email if the mnemonic changed 
we only may need to handle the validation of email in this case

Commit:
- Remove email checks
- Handle the validation of the mnemonic input directly on the mnemonic field

ramezsaeed commented:
You can also have a look here for more testing scenarios about email.
https://github.com/threefoldtech/tfgrid-sdk-ts/blob/development/packages/playground/tests/frontend_selenium/tests/TFChain/test_homepage.py

0oM4R commented:
[fix: clear email input and validation when mnemonic changes #4382](https://github.com/threefoldtech/tfgrid-sdk-ts/pull/4382)
handled here
